### PR TITLE
Update typescript types

### PR DESCRIPTION
**Pull request by AndyBitz**
Updates the types for typescript.

Is required for https://github.com/zeit/now/pull/2926

### CODE REVIEW

File: node-file-trace.d.ts [hunk 1]
```diff
@@ -1,14 +1,57 @@
...
+    type: string;
+    ignored: boolean;
+    parents: string[];
+  };
+}
+
+export interface NodeFileTraceResult {
+  fileList: string[];
+  esmFileList: string[];
+  reasons: NodeFileTraceReasons;
```
Comment (styfle) [inline node-file-trace.d.ts]: You're missing `warnings: Error[];` here

File: node-file-trace.d.ts [hunk 2]
```diff
@@ -1,14 +1,57 @@
...
+}
 
-export = NodeFileTrace;
```
Comment (styfle) [inline node-file-trace.d.ts]: Why was this changed to `default`?

Comment (AndyBitz) [reply node-file-trace.d.ts]: I've used `export = ` before, then linked it and typescript complained when I've tried to build a package

Comment (styfle) [reply node-file-trace.d.ts]: I believe `esModuleInterop` should take care of it (I think it supports both syntax options to export). But if it was failing in `@now/next`, then we can change it.

**Comment by styfle** [comment]
@AndyBitz Looks good! Feel free to release (probably semver patch?) after CI passes.

File: node-file-trace.d.ts (hunk 2 above)
Comment (AndyBitz) [reply node-file-trace.d.ts]: yup, it failed in `@now/next`
